src/controllers/organizationController.ts

```diff
@@ -414,6 +383,86 @@ export default class OrganizationController {
     return none;
   }
 
+  private createOnboardingDocuments(
+    ipaCode: string,
+    requestsCreationResponseSuccess: IResponseSuccessCreation<RequestCollection>
+  ): TaskEither<
+    IResponseErrorInternal,
+    IResponseSuccessCreation<RequestCollection>
+  > {
+    // TODO:
+    //  the documents must be stored on cloud (Azure Blob Storage).
+    //  @see https://www.pivotaltracker.com/story/show/169644958
+    const outputFolder = `./documents/${ipaCode}`;
+    const createDocumentPerRequest = (request: Request) => {
+      const rejectError = (maybeError: Option<Error>) => {
+        if (isSome(maybeError)) {
+          return Promise.reject(maybeError.value);
+        }
+      };
+      return tryCatch<IResponseErrorInternal, undefined>(
+        () => {
+          if (OrganizationRegistrationRequest.is(request)) {
+            return this.documentService
+              .generateDocument(
+                localeIt.organizationController.registerOrganization.contract.replace(
+                  "%s",
+                  `${request.organization.ipa_code} ${request.organization.fiscal_code}`
+                ),
+                `${outputFolder}/${request.document_id}.pdf`
+              )
+              .then(rejectError);
```

Comment:
do not call `.then(rejectError)`, just return `left(Error...)`

maybe something like this (I haven't tried it, just to get the idea):

```ts
return tryCatch(() => 

 // this is far from ideal as generateDocument returns an Option 
 // so you have to convert it to an error using either
 // a better way is to always return an either (instead of an option) from sync methods
 // and always return a taskEither from async methods (so yo can chain them seamlessy)

  OrganizationRegistrationRequest.is(request)) ?
    fromOption(Error("..."))(await this.documentService.generateDocument(...)) :
  UserDelegationRequest.is(request) ?
    fromOption(Error("..."))(await this.documentService.generateDocument(...)):
  left(Error("Wrong data")),

       (error: Error) =>
          genericInternalUnknownErrorHandler(
            error,
            "organizationController#createOnboardingDocuments | An error occurred during document generation.",
            "An error occurred during document generation."
          )
      );
```